service/service.go

```diff
@@ -137,6 +162,20 @@ func New(config Config) (*Service, error) {
 // Boot starts top level service implementation.
 func (s *Service) Boot() {
 	s.bootOnce.Do(func() {
-		go s.releaseController.Boot(context.Background())
+		ctx := context.Background()
+		backOffFactory := func() backoff.Interface { return backoff.NewMaxRetries(7, 1*time.Second) }
+
+		// Install Release CRD.
+		s.logger.LogCtx(ctx, "level", "debug", "message", "ensuring custom resource definition exists")
+
+		err := s.crdClient.EnsureCreated(ctx, s.crd, backOffFactory())
```

Comment:
That is not right. Why are we doing this here? The controller does it already. We should not do it manually.

Reply:
This is right. The controller creates ReleaseCycle CRD. This here is Release CRD.

Reply:
Here we are installing the `ReleaseCRD` while the controller installs the `ReleaseCycleCRD`.

I did not fond another way to have the operator install a second CRD.

Reply:
The way to install a CRD in our case is to boot a controller. Why aren't we doing simply this?

Reply:
Because the operator doesn't actually need this second controller, so it feels weird to boot a whole controller that we are not going to use.

Reply:
When the operator does not watch the CR it should not manage the CRD. Simple as that. When the Release CRD is a common resource we expect to be there it might should be bootstrapped by the control plane.

Reply:
This operator do reconciles `Release` CR and `ReleaseCycle` CR.

It watches `ReleaseCycle` CR and ensure corresponding `Release` CR are installed.
Then reconciles `Release` CR to have their components installed.

So release-operator is responsible for `Release` CRD, in the sense that is do the reconcile `Release` CR.
It would feel strange to have "something" else installing the `Release` CRD.

Reply:
Sounds like you need a controller so I am not sure what we are talking about here.

Reply:
For the record. We had a quick hang with Theo. Creating exceptions here smells. The idea now was to have a controller for the `Release` CRs and have a `releasestatus` resource reconciled in order to manage the CR status. The controller ensures the `Release` CRD. That is straight forward, clean and simple. `Release` and `ReleaseCycle` need to cross reference their 1 to 1 relationship though, which should not be hard. In case you would want to separate the controllers later on it would be as easy as moving files around. No need to deal with resource and test refactorings.

Reply:
I don't see a point in running a watcher running for Release CRs. It is an overkill for just creating a CRD.

Reply:
It is for updating the CR status separately. I think this makes sense and is a clean approach without ugly hacks and exceptions.

Reply:
We still need to watch ReleaseCycle CRs to update status. There is a reference from ReleaseCycle CR to Release CR but no other way around.